gateway/hydrator.go

```diff
@@ -230,3 +230,34 @@ func (eph *EndpointHydrator) getHealthStatus(successfulServiceChecks *sync.Map)
 
 	return true
 }
+
+func (eph *EndpointHydrator) observeReqRes(
```

Comment:
TOL (Thinking Out Loud): What do you think of renaming this to `goObserveReqRes` (everywhere) so we inform (not enforce) enforce the caller to always call it via `go eph.goObserveReqRes()`.

I don't love this idea but curious to hear this opinion. I think you have a better sense for idiomatic golang.